src/main/internal-lifecycle.js

```diff
@@ -1,52 +1,29 @@
 // @flow
 
-import invariant from "invariant";
 import { app, ipcMain } from "electron";
-import debounce from "lodash/debounce";
 import path from "path";
 import rimraf from "rimraf";
-import cluster from "cluster";
 import { setEnvUnsafe, getAllEnvs } from "@ledgerhq/live-common/lib/env";
 import { isRestartNeeded } from "~/helpers/env";
 import logger from "~/logger";
-import { setInternalProcessPID } from "./terminator";
 import { getMainWindow } from "./window-lifecycle";
-import { isTerminated } from "~/main/terminator";
+import InternalProcess from "./InternalProcess";
 
 // ~~~ Local state that main thread keep
 
 const hydratedPerCurrency = {};
-let internalProcess;
 
 // ~~~
 
 const LEDGER_CONFIG_DIRECTORY = app.getPath("userData");
 const LEDGER_LIVE_SQLITE_PATH = path.resolve(app.getPath("userData"), "sqlite");
 
+const internal = new InternalProcess(2000);
```

Comment:
would be more readable if it's a config object to explicitly say `{ timeout: 2000`